### PR TITLE
Adding WebGL BLS signature verification via JS

### PR DESCRIPTION
**Pull request by Gekctek**

### CODE REVIEW

File: Assets/WebGLTemplates/BoomTemplate/index.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <title>{{{ PRODUCT_NAME }}}</title>
     <link rel="shortcut icon" href="TemplateData/favicon.ico">
     <link rel="stylesheet" href="TemplateData/style.css">
+    <script src="TemplateData/bls.js"></script>
```
Comment (Gekctek) [inline Assets/WebGLTemplates/BoomTemplate/index.html:10]: This is required, or something like this to load the bls JS and be used from the C#/jslib

File: Assets/Scripts/Candid/CandidApiManager.cs [hunk 2]
```diff
@@ -142,8 +148,13 @@ public async UniTaskVoid CreateAgentUsingIdentityJson(string json, bool useLocal
 
                 var httpClient = new UnityHttpClient();
 
-                if (useLocalHost) await InitializeCandidApis(new HttpAgent(identity, new Uri("http://localhost:4943")));
-                else await InitializeCandidApis(new HttpAgent(httpClient, identity));
+#if UNITY_WEBGL && !UNITY_EDITOR
+                var bls = new WebGlBlsCryptography();
```
Comment (Gekctek) [inline Assets/Scripts/Candid/CandidApiManager.cs:152]: Use webgl version when necessary

File: Assets/Scripts/Candid/CandidApiManager.cs [hunk 3]
```diff
@@ -58,13 +59,18 @@ IAgent CreateAgentWithRandomIdentity(bool useLocalHost = false)
                 IAgent randomAgent = null;
 
                 var httpClient = new UnityHttpClient();
+#if UNITY_WEBGL && !UNITY_EDITOR
```
Comment (Gekctek) [inline Assets/Scripts/Candid/CandidApiManager.cs:62]: Use webgl version when necessary

File: Assets/Plugins/ICP.Net/WebGlBlsCryptography.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public bool VerifySignature(byte[] publicKey, byte[] messageHash, byte[] signature)
+    {
+        return BrowserBlsLib.VerifySignature(publicKey, messageHash, signature);
+    }
+}
+
+internal static class BrowserBlsLib
+{
+    [DllImport("__Internal")]
+    public static extern bool VerifySignature(byte[] publicKey, byte[] messageHash, byte[] signature);
```
Comment (Gekctek) [inline Assets/Plugins/ICP.Net/WebGlBlsCryptography.cs:25]: jslib connector

File: Assets/Plugins/ICP.Net/WebGlBlsCryptography.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.Networking;
+
+#if UNITY_WEBGL && !UNITY_EDITOR
+public class WebGlBlsCryptography : IBlsCryptography
```
Comment (Gekctek) [inline Assets/Plugins/ICP.Net/WebGlBlsCryptography.cs:14]: Wrapper to implement interface

File: Assets/Plugins/ICP.Net/Bls.jslib [hunk 6]
```diff
@@ -0,0 +1,6 @@
+var BLS = {
+  VerifySignature: function (publicKey, messageHash, signature) {
+    return blsVerify(publicKey, messageHash, signature);
```
Comment (Gekctek) [inline Assets/Plugins/ICP.Net/Bls.jslib:3]: calls the blsVerify JS, but requires the bls.js script to be loaded in the html/browser

File: Assets/WebGLTemplates/BoomTemplate/index.html (hunk 1 above)
Comment (Gekctek) [reply Assets/WebGLTemplates/BoomTemplate/index.html:10]: BUT, only is needed with WebGL versions. Not sure if this can be only included with WebGL builds

**Comment by Gekctek** [comment]
@tommygames just a reminder this is here
i am in no need of this, just want to make sure it doesnt get lost
Just need a review/test of the js BLS work

**Review by hedgehoghash — APPROVED** [review]

**Pull request merged**